### PR TITLE
feat(ENV-4254): adding service-catalog

### PR DESCRIPTION
**Pull request by dariapaster**
## Description

This PR addresses [feat(ENV-4254)](https://velocitytech.atlassian.net/browse/ENV-4254): Service Shop (Catalog) should be originated in Github:
In this PR I moved services data from our DB to Github.
I've created a new folder service-catalog with service data.

### CODE REVIEW

File: service-catalog/generic-service.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  name: generic-service-var-service-id
+spec:
+  ports:
+    - name: tcp
+      port: 3005
+      protocol: TCP
+      targetPort: 3000
+  selector:
+    name: generic-service-var-service-id
+  type: ClusterIP
```
Comment (IamShobe) [inline service-catalog/generic-service.yaml:45]: lets delete the service - it doesn't do anything

File: service-catalog/generic-service.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    spec:
+      containers:
+        - image: busybox:1.28
+          command:
+            - sleep
+            - "3600"
+          ports:
+            - name: http
+              containerPort: 3000
+              protocol: TCP
```
Comment (IamShobe) [inline service-catalog/generic-service.yaml:28]: delete that as well 🙏

File: service-catalog/generic-service.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    velocity.tech.v1/id: generic-service-var-service-id
+  name: generic-service-var-service-id
+spec:
+  replicas: 1
+  revisionHistoryLimit: 10
```
Comment (IamShobe) [inline service-catalog/generic-service.yaml:9]: redundant delete as well 🙏

File: service-catalog/generic-service.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  template:
+    metadata:
+      labels:
+        name: generic-service-var-service-id
+    spec:
+      containers:
+        - image: busybox:1.28
+          command:
+            - sleep
+            - "3600"
```
Comment (IamShobe) [inline service-catalog/generic-service.yaml:24]: change to `infinity`
make sure the container is running 🙏

File: service-catalog/mongo.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+apiVersion: apps/v1
```
Comment (IamShobe) [inline service-catalog/mongo.yaml:1]: can you please run https://github.com/itaysk/kubectl-neat
on all files to remove redundant lines? 🙏 thanks :)

File: service-catalog/mysql.yaml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    metadata:
+      labels:
+        name: mysql-database-var-service-id
+    spec:
+      containers:
+        - env:
+            - name: MYSQL_DATABASE
+              value: app_development
+            - name: MYSQL_PASSWORD
+              value: password123
```
Comment (IamShobe) [inline service-catalog/mysql.yaml:29]: lets replace with generate random annotation

File: service-catalog/mysql.yaml [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            - name: MYSQL_ROOT_PASSWORD
+              value: 8QeO1qaL
```
Comment (IamShobe) [inline service-catalog/mysql.yaml:31]: lets replace with generate random annotation

File: service-catalog/mysql.yaml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+            - name: MYSQL_USER
+              value: user123
```
Comment (IamShobe) [inline service-catalog/mysql.yaml:33]: lets replace with generate random annotation

File: service-catalog/mongo.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        name: mongodb-var-service-id
+    spec:
+      containers:
+        - env:
+            - name: MONGODB_ROOT_PASSWORD
+              value: "password1234"
```
Comment (IamShobe) [inline service-catalog/mongo.yaml:26]: lets replace with generate random annotation

File: service-catalog/postgres.yaml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        name: postgres-database-var-service-id
+    spec:
+      containers:
+        - env:
+            - name: POSTGRESQL_DATABASE
+              value: app_development
+            - name: POSTGRESQL_PASSWORD
+              value: secretpass
+            - name: POSTGRESQL_USERNAME
+              value: username
```
Comment (IamShobe) [inline service-catalog/postgres.yaml:31]: lets replace with generate random annotation

File: service-catalog/postgres.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    metadata:
+      labels:
+        name: postgres-database-var-service-id
+    spec:
+      containers:
+        - env:
+            - name: POSTGRESQL_DATABASE
+              value: app_development
+            - name: POSTGRESQL_PASSWORD
+              value: secretpass
```
Comment (IamShobe) [inline service-catalog/postgres.yaml:29]: lets replace with generate random annotation

File: service-catalog/redis.yaml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        name: redis-var-service-id
+    spec:
+      containers:
+        - env:
+            - name: REDIS_PASSWORD
+              value: "password123"
```
Comment (IamShobe) [inline service-catalog/redis.yaml:25]: lets replace with generate random annotation

File: service-catalog/generic-service.yaml [hunk 13]
```diff
@@ -0,0 +1,45 @@
+apiVersion: apps/v1
```
Comment (IamShobe) [inline service-catalog/generic-service.yaml:1]: all those exists in: 
https://github.com/techvelocity/velocity-blueprints/tree/main/references/kubernetes/database-containers
please put it there with the changes required 🙏

**Review by IamShobe — CHANGES_REQUESTED** [review]
Well done, few changes required please 🙏

File: service-catalog/generic-service.yaml (hunk 1 above)
Comment (dariapaster) [reply service-catalog/generic-service.yaml:45]: without Service, I can't run development for  generic-service

Comment (IamShobe) [reply service-catalog/generic-service.yaml:45]: it's ok, the editor will add this service once needed.
for now, it is not being used, and for starters we need a clean slate for a service.

File: service-catalog/generic-service.yaml (hunk 4 above)
Comment (dariapaster) [reply service-catalog/generic-service.yaml:24]: `sleep infinity` doesn't work :(

Comment (IamShobe) [reply service-catalog/generic-service.yaml:24]: try a newer version of busybox then
https://hub.docker.com/_/busybox
https://stackoverflow.com/questions/63527494/why-busybox-container-is-in-completed-state-instead-of-running-state
I want it to stay alive forever if possible

File: service-catalog/generic-service.yaml (hunk 13 above)
Comment (dariapaster) [reply service-catalog/generic-service.yaml:1]: there are files with the same name. can I replace the existing ones?

File: service-catalog/generic-service.yaml (hunk 3 above)
Comment (dariapaster) [reply service-catalog/generic-service.yaml:9]: sorry, I didn't get, what exactly is redundant here. Do you mean replicas and revisionHistoryLimit?

File: service-catalog/generic-service.yaml (hunk 13 above)
Comment (IamShobe) [reply service-catalog/generic-service.yaml:1]: look at the diff in each.
make sure all are additions

File: references/kubernetes/catalog.yaml [hunk 14]
```diff
@@ -0,0 +1,13 @@
+version: 1
+services:
+  mongodb:
+    path: https://raw.githubusercontent.com/techvelocity/velocity-blueprints/main/references/kubernetes/database-containers/mongodb.yaml
```
Comment (IamShobe) [inline references/kubernetes/catalog.yaml:4]: without prefix :)

File: references/kubernetes/catalog.yaml [hunk 15]
```diff
@@ -0,0 +1,13 @@
+version: 1
```
Comment (IamShobe) [inline references/kubernetes/catalog.yaml:1]: lets put it in the root of the repo

**Review by IamShobe — APPROVED** [review]
Nice work!!! 
very good and fast 💪

**Pull request merged**